pkgs/by-name/nu/nuget-to-json/nuget-to-json.sh

```diff
@@ -0,0 +1,119 @@
+    if [[ -d "$index" ]]; then
+        continue
+    fi
+
+    remote_sources+=("$index")
+
+    base_address=$(
+        curl --compressed --netrc -fsL "$index" |
+            jq -r '.resources[] | select(."@type" == "PackageBaseAddress/3.0.0")."@id"'
+    )
+    if [[ ! "$base_address" == */ ]]; then
+        base_address="$base_address/"
+    fi
+    base_addresses[$index]="$base_address"
+done
+
+(
+    echo '['
+
+    first=true
+    cd "$pkgs"
+    for package in *; do
+        [[ -d "$package" ]] || continue
+        cd "$package"
+        for version in *; do
+            id=$(xmlstarlet sel -t -v /_:package/_:metadata/_:id "$version"/*.nuspec)
+
+            if grep -qxF "$id.$version.nupkg" "$excluded_list"; then
+                continue
+            fi
+
+            # packages in the nix store should have an empty metadata file
+            used_source="$(jq -r '.source // ""' "$version"/.nupkg.metadata)"
+            found=false
+
+            if [[ -z "$used_source" || -d "$used_source" ]]; then
+                continue
+            fi
+
+            for source in "${remote_sources[@]}"; do
+                url="${base_addresses[$source]}$package/$version/$package.$version.nupkg"
+                if [[ "$source" == "$used_source" ]]; then
+                    hash="$(nix-hash --type sha256 --flat --sri "$version/$package.$version".nupkg)"
+                    found=true
+                    break
+                else
+                    if hash=$(nix-prefetch-url "$url" 2>"$tmp"/error); then
+                        hash="$(nix-hash --to-sri --type sha256 "$hash")"
+                        # If multiple remote sources are enabled, nuget will try them all
+                        # concurrently and use the one that responds first. We always use the
+                        # first source that has the package.
+                        echo "$package $version is available at $url, but was restored from $used_source" 1>&2
+                        found=true
+                        break
+                    else
+                        if ! grep -q 'HTTP error 404' "$tmp/error"; then
+                            cat "$tmp/error" 1>&2
+                            exit 1
+                        fi
+                    fi
+                fi
+            done
+
+            if [[ $found = false ]]; then
+                echo "couldn't find $package $version" >&2
+                exit 1
+            fi
+
+            if [[ $first = false ]]; then
+                echo '  , {'
+            else
+                echo '  {'
+            fi
+            echo "    \"pname\": \"$id\""
+            echo "    , \"version\": \"$version\""
+            echo "    , \"hash\": \"$hash\""
+            if [[ "$source" != https://api.nuget.org/v3/index.json ]]; then
+                echo "    , \"url\": \"$url\""
+            fi
+            echo '  }'
+            first=false
+        done
+        cd ..
+    done
+
+    echo ']'
+) | jq .
```

Comment:
No idea, here I'm only using it to format the JSON and it doesn't seem to reorder the object properties so it should be good enough for our use case I think.

Reply:
No idea, here I'm only using it to format the JSON and it doesn't seem to reorder the object properties so it should be good enough for our use case I think.

Reply:
Yeah, I don't see any thing concerning in the docs.  It would be a shame if they drastically changed the output all of a sudden, but that seems unlikely.

There is something in the docs for `debug`:

> where <input-value> is a compact rendition of the input value. *This format may change in the future.*

Which indicates that they are at least conscious enough of output potentially changing to warn about it.